caching-big-repositories_5bf03fb92c7d3a31944e139d.md

```diff
@@ -0,0 +1,105 @@
+
+* [Overview](#overview)
+* [The logic behind the recipe](#the-logic-behind-the-recipe)
+* [An example project](#an-example-project)
+* [See Also](#see-also)
+
+## Overview
+
+Imagine that you have one or multiple large files on a GitHub repository and
+you want to make the downloading of them as fast as possible.
+
+Note that for this recipe to work, the files of the GitHub repository should
+remain intact through the lifetime of the Semaphore project.
```

Comment:
I'm surprised by this. Are you sure?

Reply:
I should have said `lifetime of pipeline` instead.

Reply:
I honestly can't understand this.
What is the message this paragraph is conveying?
What does it mean "repository should remain intact"?
What is the scenario that users need to protect themselves against and how?